kratos/processes/calculate_distance_to_skin_process.cpp

```diff
@@ -24,278 +24,285 @@
 namespace Kratos
 {
 
-	template<std::size_t TDim>
-	CalculateDistanceToSkinProcess<TDim>::CalculateDistanceToSkinProcess(
-		ModelPart& rVolumePart,
-		ModelPart& rSkinPart)
-		: CalculateDiscontinuousDistanceToSkinProcess<TDim>(rVolumePart, rSkinPart)
-	{
-	}
-
-	template<std::size_t TDim>
-	CalculateDistanceToSkinProcess<TDim>::CalculateDistanceToSkinProcess(
-		ModelPart& rVolumePart,
-		ModelPart& rSkinPart,
-		const double RayCastingRelativeTolerance)
-		: CalculateDiscontinuousDistanceToSkinProcess<TDim>(rVolumePart, rSkinPart),
-		mRayCastingRelativeTolerance(RayCastingRelativeTolerance)
-	{
-	}
-
-	template<std::size_t TDim>
-	CalculateDistanceToSkinProcess<TDim>::CalculateDistanceToSkinProcess(
-		ModelPart& rVolumePart,
-		ModelPart& rSkinPart,
-		Parameters rParameters)
-		: CalculateDiscontinuousDistanceToSkinProcess<TDim>(rVolumePart, rSkinPart)
-	{
-		rParameters.RecursivelyValidateAndAssignDefaults(GetDefaultParameters());
-		mRayCastingRelativeTolerance = rParameters["ray_casting_relative_tolerance"].GetDouble();
-        CalculateDiscontinuousDistanceToSkinProcess<TDim>::mpElementalDistancesVariable = &KratosComponents<Variable<Vector>>::Get(rParameters["elemental_distances_variable"].GetString());;
-        mpDistanceVariable = &KratosComponents<Variable<double>>::Get(rParameters["distance_variable"].GetString());;
-		const std::string distance_database = rParameters["distance_database"].GetString();
-		if (distance_database == "nodal_historical") {
-			mDistanceDatabase = DistanceDatabase::NodeHistorical;
-		} else if (distance_database == "nodal_non_historical") {
-			mDistanceDatabase = DistanceDatabase::NodeNonHistorical;
-		} else {
-			KRATOS_ERROR << "Provided 'distance_database' is '" << distance_database << "'. Available options are 'nodal_historical' and 'nodal_non_historical'." <<  std::endl;
```

Comment:
Indeed, I'd use this same error message in the error thrown you're adding to the ray-casting process to keep consistency.

Reply:
I think I will do it the other way around as this message contains more info.

Reply:
Yes, this is what I meant. Using the one that contains more info.